addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -476,3 +479,38 @@ def _get_move_applicability(self, move):
             'post': self._l10n_sa_post_zatca_edi,
             'edi_content': self._l10n_sa_get_invoice_content_edi,
         }
+
+    def _prepare_invoice_report(self, pdf_writer, edi_document):
+        """
+        Prepare invoice report to be printed.
+        :param pdf_writer: The pdf writer with the invoice pdf content loaded.
+        :param edi_document: The edi document to be added to the pdf file.
+        """
+        super()._prepare_invoice_report(pdf_writer, edi_document)
+        if self.code != 'sa_zatca' or edi_document.move_id.country_code != 'SA':
+            return
+
+        attachment = edi_document.attachment_id
+        if not attachment or not attachment.datas:
+            _logger.warning(f"No attachment found for invoice {edi_document.move_id.name}")
+            return
+
+        # Convert attachment data from base64 to raw bytes.
+        xml_content = b64decode(attachment.datas)
+        file_name = attachment.name
+
+        # Embed the attachment into the PDF writer.
+        pdf_writer.addAttachment(file_name, xml_content, subtype='text/xml')
+        if not pdf_writer.is_pdfa:
+            try:
+                pdf_writer.convert_to_pdfa()
+            except Exception:
+                _logger.exception("Error while converting to PDF/A")
```

Comment:
except Exception is generally a bad practice but it seems we don't have the choice here...
May I ask you to add the exception (like we did in the UBL module) in the logger?
https://github.com/odoo/odoo/blob/9b02bcfc0497b0d3e47b9b893398fa4701433ab4/addons/account_edi_ubl_cii/models/account_edi_format.py#L169-L172